examples/tutorials/text.py

```diff
@@ -0,0 +1,128 @@
+
+fig.show()
+
+########################################################################################
+# Plotting from a text file
+# -------------------------
+#
+# It is also possible to add annotations from a file containing `x`, `y`, and
+# `text` fields. Here we give a complete example.
+
+fig = pygmt.Figure()
+with pygmt.config(MAP_FRAME_TYPE="plain"):
+    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
+fig.coast(land="black", water="skyblue")
+
+# Plot region names / sea names
+fig.text(textfiles="examples.txt", font="22p,Helvetica-Bold,white")
+
+# Plot names of seas
+fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+fig.text(text="SULU SEA", x=119.12, y=7.25, angle=-40)
+fig.text(text="SOUTH CHINA SEA", x=112, y=6, angle=40)
+fig.text(text="MAKASSAR STRAIT", x=118.4, y=-1, angle=65)
+fig.show()
+
+########################################################################################
+# ``justify`` argument
+# --------------------
+#
+# ``justify`` is used to define the anchor point for the bounding box for text
+# being added to a plot. The following code segment demonstrates the
+# positioning of the anchor point relative to the text.
+#
+# The anchor is specified with a two letter (order independent) code, chosen
+# from:
+# * Horizontal anchor: L(eft), C(entre), R(ight)
+# * Vertical anchor: T(op), M(iddle), B(ottom)
+
+fig = pygmt.Figure()
+fig.basemap(region=[0, 4, 0, 4], projection="X10c", frame="WSen")
+for i, y_justify in enumerate(["T", "M", "B"]):
+    y_pos = 3.5 - i*1.5
+    fig.plot(x=[0., 4.], y=[y_pos, y_pos], W='3p,red@85')
+    for j, x_justify in enumerate(["L", "C", "R"]):
+        justify_text = x_justify + y_justify
+        x_pos = 0.5 + j*1.5
+        fig.text(text=justify_text, x=x_pos, y=y_pos,
+                 font="28p,Helvetica-Bold,black", justify=justify_text)
+        fig.plot(x=[x_pos, x_pos], y=[0., 4.], W="3p,red@85")
+fig.show()
+
+########################################################################################
+# ``angle`` argument
+# ------------------
+# ``angle`` is an optional argument used to specify the clockwise rotation of
+# the text from the horizontal.
+
+fig = pygmt.Figure()
+fig.basemap(region=[0, 4, 0, 4], projection="X10c", frame="WSen")
+for i in range(0, 360, 30):
+    fig.text(text=f"`          {i} Degrees", x=2, y=2, justify="LM", angle=i)
+fig.show()
+
+########################################################################################
+# Additional arguments
+# --------------------
+#
+# Text can be further configured by passing an argument corresponding to the
+# flag names in GMT, following the same convention as described in the GMT
+# documentation. It is hoped that over time more bindings to these arguments
+# will be written into PyGMT.
+
+fig = pygmt.Figure()
+fig.basemap(region=[0, 1, 0, 1], projection="X5c", frame="WSen")
+fig.text(text="Green", x=0.5, y=0.5, G="green")
```

Comment:
We're actually trying to document only the long-form arguments (see #473), and have made a start in #474. Just that we haven't released it yet, see differences in documentation between https://www.pygmt.org/dev/api/generated/pygmt.Figure.text.html and https://www.pygmt.org/v0.1.1/api/generated/pygmt.Figure.text.html#pygmt.Figure.text. I've been thinking whether we should cut a v0.1.2 release to incorporate this PR and other documentation changes, before GMT 6.1.0 comes out around end of June.

Personally I have nothing against using short aliases like this to achieve additional functionality (since I do it too!). Better way would be to add an alias for G=color here, see #385 for an example on how to do so.

Reply:
Yes, I was planning to collect some thoughts on how best to bring the rest of the available flags, or short-form arguments, into the PyGMT fold for `text` and submit an issue. 

So for now, should I remove this last section from the tutorial?

Reply:
Or we could do a quick PR to get some extra aliases into `text`. They're basically one-line additions (per alias). Do you have a list of what you want besides `color`?

Reply:
It would make sense to also add an alias for W (`pen`?) if adding one for G.

Reply:
> It would make sense to also add an alias for W (`pen`?) if adding one for G.

Yes, W=pen seems like another obvious choice.

> I find having `font` as an alias for F when +f also relates to `font` a bit ambiguous. There was also the point I raised in Issue #479 - when adding text from a text file, it is unclear if it's actually possible to emulate the way you tell GMT to expect `font`, `angle`, or `justify` fields in the file (`-F+f+a+j`).

Looking at the current v0.1.1 `text` implementation https://github.com/GenericMappingTools/pygmt/blob/809a888755c5dedc86ea74d0ae0497295b5311a4/pygmt/base_plotting.py#L864-L872

I think if you want to emulate `-F+f+a+j`, you could do `fig.text(..., font=True, angle=True, justify=True), and that would then use treat the columns in the textfile as font/angle/justify respectively. Unfortunately it won't be possible to change the order of the columns (e.g. `-F+a+j+f` I think.

But yes, you're right that it's an ambiguous beast. We could either document this particular case for GMT 'power' users, or find a better way to implement the code block above, bearing in mind that PyGMT is meant to be accessible to new users (see [Project Goals](https://www.pygmt.org/latest/#project-goals)) who might not be as familiar with the GMT suite. It would be nice to somewhat emulate what [`matplotlib.pyplot.text`](https://matplotlib.org/3.2.1/api/_as_gen/matplotlib.pyplot.text.html) does for example (i.e. have a `fontdict` like argument).

Reply:
angle, font, and justify should also accept list of input, so that user can plot texts with varying angles, fonts or justification. It should be addressed in a separate issue/PR.

Reply:
I don't think that works:

```
> justify = True
> isinstance(justify, str)
False
```

etc

Reply:
Oh right, forgot that `justify` isn't an alias like `frame`, so the boolean doesn' convert to an empty string. In that case we could use `justify=""`?

Reply:
Yes - works for `justify` and `font`, but wouldn't work for angle. Ideally though you would want to be able to set them in the same way?

Reply:
To illustrate:

```python
# Create file
with open("examples.txt", "w") as f:
    print(114, 0.5, "22p,Helvetica-Bold,white", "LM", "BORNEO", file=f)

# Plot region names / sea names
fig.text(textfiles="examples.txt", font="", justify="")

# Cleanup
os.remove("examples.txt")
```

This works.

Reply:
> Yes - works for `justify` and `font`, but wouldn't work for angle. Ideally though you would want to be able to set them in the same way?

True, should allow `str` type for `angle` as well. I'll start a Pull Request to add some enhancements to `text`, give me a few minutes.

Reply:
Ok, I've started PR #481 to handle this, so `angle=""` should work once it's merged in to master. Also checked that `angle=True`, `font=True` and `justify=True` works (See test case at https://github.com/GenericMappingTools/pygmt/pull/481/files#diff-c20c076528e6988bc916b4bfc47e7a8d.
).  The documentation at https://www.pygmt.org/dev/api/generated/pygmt.Figure.text does mention that `bool` types are allowed. That would translate directly to `-F+a+f+j` to GMT.

Reply:
In which case, perhaps it would be best to wait until the above has been merged, so we can give the example as:

```python
fig = pygmt.Figure()
with pygmt.config(MAP_FRAME_TYPE="plain"):
    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
fig.coast(land="black", water="skyblue")

# Create file
with open("examples.txt", "w") as f:
    f.write("114 0.5 22p,Helvetica-Bold,white 0 CM BORNEO\n")
    f.write("119 3.25 12p,Helvetica-Bold,black 0 CM CELEBES SEA\n")
    f.write("112 -4.6 12p,Helvetica-Bold,black 0 CM JAVA SEA\n")
    f.write("112 6 12p,Helvetica-Bold,black 40 CM SOUTH CHINA SEA\n")
    f.write("119.12 7.25 12p,Helvetica-Bold,black -40 CM SULU SEA\n")
    f.write("118.4 -1 12p,Helvetica-Bold,black 65 CM MAKASSAR STRAIT\n")

# Plot region names / sea names
fig.text(textfiles="examples.txt", angle=True, justify=True, font=True)

# Cleanup
os.remove("examples.txt")

fig.show()
```

Reply:
Ok, we've merged in #481! If you can merge in the new changes from master into this branch (e.g. by clicking on the 'Update Branch' button), the new features/enhancement should be available and we can simplify some of the example code. Let us know if you need any help.

Reply:
The example might be another reason to support `textfile=io.StringIO` input as with #576, just so we don't need to explicitly create a temporary .txt file. But let's do this another time :wink: